plugins/woocommerce-blocks/tests/e2e/tests/local-pickup/local-pickup.merchant.block_theme.spec.ts

```diff
@@ -0,0 +1,292 @@
+		await expect(
+			page.locator( '#inspector-checkbox-control-0' )
+		).toBeChecked();
+	} );
+
+	test.describe( 'Core Settings', () => {
+		test.beforeEach( async ( { admin, page } ) => {
+			await utils.clearLocations( admin, page );
+			await utils.removeCostForLocalPickup( admin, page );
+			await utils.enableLocalPickup( admin, page );
+		} );
+
+		test( 'Shows the correct shipping options depending on whether Local Pickup is enabled', async ( {
+			admin,
+			page,
+		} ) => {
+			await utils.disableLocalPickup( admin, page );
+
+			await admin.visitAdminPage(
+				'admin.php',
+				'page=wc-settings&tab=shipping&section=pickup_location'
+			);
+
+			await expect(
+				page.locator( '#inspector-checkbox-control-0' )
+			).not.toBeChecked();
+
+			await admin.visitAdminPage(
+				'admin.php',
+				'page=wc-settings&tab=shipping&section=options'
+			);
+
+			expect(
+				page.locator( '#woocommerce_shipping_cost_requires_address' )
+			).not.toBeNull();
+
+			await utils.enableLocalPickup( admin, page );
+
+			await admin.visitAdminPage(
+				'admin.php',
+				'page=wc-settings&tab=shipping&section=options'
+			);
+
+			expect(
+				page.locator( '#woocommerce_shipping_cost_requires_address' )
+			).not.toBeNull();
+		} );
+	} );
+
+	test.describe( 'Global Settings', () => {
+		test.beforeEach( async ( { admin, page } ) => {
+			await utils.clearLocations( admin, page );
+			await utils.removeCostForLocalPickup( admin, page );
+			await utils.enableLocalPickup( admin, page );
+		} );
+
+		test( 'Allows toggling of enabled state', async ( { admin, page } ) => {
+			await admin.visitAdminPage(
+				'admin.php',
+				'page=wc-settings&tab=shipping&section=pickup_location'
+			);
+
+			const text = 'Enable local pickup';
+			const xpath = `//label[text()='${ text }']/preceding-sibling::span`;
+			const checkedState = page
+				.locator( xpath )
+				.locator( 'svg' )
+				.isVisible();
+
+			await page.getByText( text ).click();
+			await utils.savelocalPickupSettings( admin, page );
+
+			const newCheckedState = page
+				.locator( xpath )
+				.locator( 'svg' )
+				.isVisible();
+
+			expect( newCheckedState ).not.toBe( checkedState );
+		} );
+
+		test( 'Allows the title to be changed', async ( { admin, page } ) => {
+			await admin.visitAdminPage(
+				'admin.php',
+				'page=wc-settings&tab=shipping&section=pickup_location'
+			);
+
+			await page
+				.locator( 'input[name="local_pickup_title"]' )
+				.fill( 'Local Pickup Test #1' );
+
+			await utils.savelocalPickupSettings( admin, page );
+
+			await expect(
+				page.locator( 'input[name="local_pickup_title"]' )
+			).toHaveValue( 'Local Pickup Test #1' );
+
+			await page
+				.locator( 'input[name="local_pickup_title"]' )
+				.fill( 'Local Pickup Test #2' );
+
+			await utils.savelocalPickupSettings( admin, page );
+
+			await expect(
+				page.locator( 'input[name="local_pickup_title"]' )
+			).toHaveValue( 'Local Pickup Test #2' );
```

Comment:
```suggestion
			const changeTitleAndVerify = async ( title: string ) => {
				await page.getByPlaceholder( 'Local Pickup' ).fill( title );
				await utils.savelocalPickupSettings( admin, page );
				await expect( page.getByText( title ) ).toBeTruthy();
			};

			await changeTitleAndVerify( 'Local Pickup Test #1' );

			await changeTitleAndVerify( 'Local Pickup Test #2' );
```

What are your thoughts on introducing a helper function for filling, saving and checking the title? That way, the test is more concise and easier to read.